sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardReadersPool.java

```diff
@@ -44,13 +48,14 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(ShardReadersPool.class);
   private static final int DEFAULT_CAPACITY_PER_SHARD = 10_000;
-  private ExecutorService executorService;
-  private BlockingQueue<KinesisRecord> recordsQueue;
-  private Map<String, ShardRecordsIterator> shardIteratorsMap;
-  private SimplifiedKinesisClient kinesis;
-  private KinesisReaderCheckpoint initialCheckpoint;
+  private final ExecutorService executorService;
+  private volatile BlockingQueue<KinesisRecord> recordsQueue;
```

Comment:
Why adding volatiles?

Reply:
These fields are accessed by multiple threads and synchronization is used only during their update to guarantee atomicity. Volatility guarantees that in all other places the most recent value is accessed. IMO the code looks better here with volatiles than atomic references.

Reply:
I would prefer atomic references, and some comments here about how these are accessed (who reads and modifies them).